src/BcPrebidVast.js

```diff
@@ -636,11 +641,76 @@ function loadMolPlugin(callback) {
     }
 }
 
+function getAdRendererFromAdOptions(adOptions) {
+	// if adRenderer option is present use it for all ads
+	if (adOptions.hasOwnProperty('adRenderer') && adOptions.adRenderer &&
+		(adOptions.adRenderer === _rendererNames.MOL || adOptions.adRenderer === _rendererNames.IMA)) {
+		return {adRenderer: adOptions.adRenderer, needBreak: true};
+	}
+	// for DFP use IMA plugin as renderer
+	if (adOptions.hasOwnProperty('dfpParameters')) {
+		return {adRenderer: _rendererNames.IMA, needBreak: false};
+	}
+	return null;
+}
+
+function setAdRenderer(options) {
+	if (options) {
+		var adRenderer = _rendererNames.MOL;
+		var rendObj;
+		var i;
+		if (Array.isArray(options)) {
+			for (i = 0; i < options.length; i++) {
+				rendObj = getAdRendererFromAdOptions(options[i]);
+				if (rendObj) {
+					adRenderer = rendObj.adRenderer;
+					if (rendObj.needBreak) {
+						break;
+					}
+				}
+			}
+			if (adRenderer) {
+				for (i = 0; i < options.length; i++) {
+					options[i].adRenderer = adRenderer;
+				}
+			}
+		}
+		else {
+			// array in brightcove studio converted to object {0: {...}, 1: {...}, ...}
+			if (options.hasOwnProperty('0')) {
+				// options parameter is array of options from plugin embedded in player in studio
+				for (i = 0; options.hasOwnProperty(i); i++) {
+					rendObj = getAdRendererFromAdOptions(options[i]);
+					if (rendObj) {
+						adRenderer = rendObj.adRenderer;
+						if (rendObj.needBreak) {
+							break;
+						}
+					}
+				}
+				for (i = 0; options.hasOwnProperty(i); i++) {
+					options[i].adRenderer = adRenderer;
+				}
+			}
+			else {
+				rendObj = getAdRendererFromAdOptions(options);
+				if (rendObj) {
```

Comment:
Can we add a 2nd condition in the if (to make sure the adRenderer isn't already set), and a comment here about why this case may happen (we found dfp settings but not pre-set adRenderer in the options)?

`if(rendObj && !options.adRenderer) {`

Reply:
No. Old code protects from fake renderer name.